dsl/pace/dsl/dace/utils.py

```diff
@@ -242,5 +261,112 @@ def count_memory(sdfg: dace.sdfg.SDFG, detail_report=False) -> str:
+      for higher accuracy. Best is to run a copy_stencils on a full domain
+    - Memory pressure is mostly in read/write from global memory, inner scalar & shared
+      memory is not counted towards memory movement.
+    """
+    allmaps = [
+        (me, state)
+        for me, state in sdfg.all_nodes_recursive()
+        if isinstance(me, dace.nodes.MapEntry)
+    ]
+    topmaps = [
+        (me, state) for me, state in allmaps if get_parent_map(state, me) is None
+    ]
+
+    result: Dict[str, float] = {}
+    for node, state in topmaps:
+        nsdfg = state.parent
+        mx = state.exit_node(node)
+
+        # Gather all memory read & write by reading all
+        # in-node & out-node memory. All in bytes
+        alldata_in_bytes = sum(
+            [
+                dace.data._prod(e.data.subset.size())
+                * nsdfg.arrays[e.data.data].dtype.bytes
+                for e in state.in_edges(node)
+            ]
+        )
+        alldata_in_bytes += sum(
+            [
+                dace.data._prod(e.data.subset.size())
+                * nsdfg.arrays[e.data.data].dtype.bytes
+                for e in state.out_edges(mx)
+            ]
+        )
+
+        # Compute hardware memory bandwith in bytes/us
+        if hardware not in _HARDWARE_BW_GB_S.keys():
+            print(
+                f"Timing analysis: hardware {hardware} unknown"
+                " reading hardware_bw_in_Gb_s option"
+            )
+            bandwidth_in_bytes_s = hardware_bw_in_Gb_s * 1024 * 1024 * 1024
+        else:
+            # Time it has to take (at least): bytes / bandwidth_in_bytes_s
+            bandwidth_in_bytes_s = _HARDWARE_BW_GB_S[hardware] * 1024 * 1024 * 1024
+        in_us = 1000 * 1000
+
+        # Theoritical fastest timning
+        try:
+            newresult_in_us = (float(alldata_in_bytes) / bandwidth_in_bytes_s) * in_us
+        except TypeError:
+            newresult_in_us = (alldata_in_bytes / bandwidth_in_bytes_s) * in_us
+
+        if node.label in result:
+            import sympy
+
+            newresult_in_us = sympy.Max(result[node.label], newresult_in_us).expand()
+            try:
+                newresult_in_us = float(newresult_in_us)
+            except TypeError:
+                pass
+
+        # Bad expansion
+        if not isinstance(newresult_in_us, float):
+            continue
+
+        result[node.label] = newresult_in_us
+
+    return result
+
+
+def report_kernel_theoritical_timing(
+    timings: Dict[str, float], human_readable: bool = True, csv: bool = False
+):
+    """Produce a human readable or CSV of the kernel timings"""
+    result_string = f"Maps processed: {len(timings)}.\n"
+    if human_readable:
+        result_string += "Timing in microseconds  Map name:\n"
+        result_string += "\n".join(f"{v:.2f}\t{k}," for k, v in sorted(timings.items()))
+    if csv:
+        result_string += "#Map name,timing in microseconds\n"
```

Comment:
do we want white-spaced keys in csv? I'm generally avoiding it but am fine if you like it

Reply:
I didn't use too, but it's legal and it reads a bit better so `-_o_-`